src/components/PrimaryPanes/Outline.js

```diff
@@ -39,6 +53,46 @@ export class Outline extends Component<Props> {
     selectLocation({ sourceId: selectedSourceId, line: startLine });
   }
 
+  onContextMenu(event: SyntheticEvent<HTMLElement>, func: SymbolDeclaration) {
+    event.stopPropagation();
+    event.preventDefault();
+
+    const {
+      selectedSource,
+      getFunctionText,
+      getFunctionLocation,
+      flashLineRange,
+      selectedLocation
+    } = this.props;
+
+    const copyFunctionKey = L10N.getStr("copyFunction.accesskey");
+    const copyFunctionLabel = L10N.getStr("copyFunction.label");
+
+    if (!selectedSource) {
+      return;
+    }
+
+    const sourceLine = func.location.start.line;
+    const functionText = getFunctionText(sourceLine);
+    const copyFunctionItem = {
+      id: "node-menu-copy-function",
+      label: copyFunctionLabel,
+      accesskey: copyFunctionKey,
+      disabled: !functionText,
+      click: () => {
+        const { location: { start, end } } = getFunctionLocation(sourceLine);
+        flashLineRange({
+          start: start.line,
+          end: end.line,
+          sourceId: selectedLocation.sourceId
+        });
+        return copyToTheClipboard(functionText);
+      }
+    };
+    const menuOptions = [copyFunctionItem];
+    showMenu(event, menuOptions);
```

Comment:
In some other code i saw dispatching action instead of imperative call, not sure which is right?

Reply:
This looks good to me :)